app/controllers/api/v1/locations_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::V1::LocationsController < ApplicationController
+
+  def show
+    location = Location.find_by(ethereum_address: params[:id])
+    if location
+      render json: location
+    else
+      head 404
+    end
+  end
+
+  def create
+    location = Location.new(location_params)
+    if location.save
+      head 201
```

Comment:
@DavidKnott Is it sufficient to return only the HTTP status code or would there be a reason to return the object as well?